account_payment_mode_auto_reconcile/models/account_invoice.py

```diff
@@ -82,6 +84,16 @@ def auto_reconcile_credits(self, partial_allowed=True):
                     continue
                 invoice.assign_outstanding_credit(credit.get('id'))
 
+    @api.multi
+    def _filter_payment_same_journal(self, credits_dict):
+        """Keep only credits on the same journal than the invoice."""
+        self.ensure_one()
+        line_ids = [credit['id'] for credit in credits_dict]
+        lines = self.env['account.move.line'].browse(line_ids).filtered(
+            lambda line: line.journal_id == self.journal_id
+        )
```

Comment:
@TDu To get better performance (don't know how many records we could process here), could you rewrite this using `search` instead of `browse`+`filtered`? That way no prefetch will be done on the database as you are also only using `line.ids` right after.

Reply:
Thanks for your review, no compromise on performance, fixed.